runelite-client/src/main/java/net/runelite/client/plugins/driftnet/DriftNetPlugin.java

```diff
@@ -0,0 +1,293 @@
+
+	@Inject
+	private DriftNetOverlay overlay;
+
+	@Getter
+	private Set<NPC> fish = new HashSet<>();
+	@Getter
+	private Map<NPC, Integer> taggedFish = new HashMap<>();
+	@Getter
+	private final List<DriftNet> NETS = ImmutableList.of(
+		new DriftNet(NullObjectID.NULL_31433, Varbits.NORTH_NET_STATUS, Varbits.NORTH_NET_CATCH_COUNT),
+		new DriftNet(NullObjectID.NULL_31434, Varbits.SOUTH_NET_STATUS, Varbits.SOUTH_NET_CATCH_COUNT));
+
+	@Getter
+	private boolean inDriftNetArea;
+	private boolean armInteraction;
+
+	@Provides
+	DriftNetConfig provideConfig(ConfigManager configManager)
+	{
+		return configManager.getConfig(DriftNetConfig.class);
+	}
+
+	@Override
+	protected void startUp()
+	{
+		overlayManager.add(overlay);
+
+		if (client.getGameState() == GameState.LOGGED_IN)
+		{
+			clientThread.invokeLater(() ->
+			{
+				inDriftNetArea = checkArea();
+				updateDriftNetVarbits();
+			});
+		}
+	}
+
+	@Override
+	protected void shutDown()
+	{
+		overlayManager.remove(overlay);
+		fish.clear();
+		taggedFish.clear();
+		armInteraction = false;
+	}
+
+	@Subscribe
+	public void onGameStateChanged(GameStateChanged event)
+	{
+		switch (event.getGameState())
+		{
+			case LOGIN_SCREEN:
+			case HOPPING:
+				fish.clear();
+				taggedFish.clear();
+				break;
+			case LOGGED_IN:
+			case LOADING:
+				inDriftNetArea = checkArea();
+				updateDriftNetVarbits();
+				break;
+		}
+	}
+
+	@Subscribe
+	public void onVarbitChanged(VarbitChanged event)
+	{
+		updateDriftNetVarbits();
+	}
+
+	private void updateDriftNetVarbits()
+	{
+		if (!inDriftNetArea)
+		{
+			return;
+		}
+
+		for (DriftNet net : NETS)
+		{
+			DriftNetStatus status = DriftNetStatus.of(client.getVar(net.getStatusVarbit()));
+			int count = client.getVar(net.getCountVarbit());
+
+			net.setStatus(status);
+			net.setCount(count);
+		}
+
+		// When you collect any loot, all tags become invalidated
+		if (client.getVar(Varbits.DRIFT_NET_COLLECT) != 0)
+		{
+			taggedFish.clear();
+		}
+	}
+
+	@Subscribe
+	public void onInteractingChanged(InteractingChanged event)
+	{
+		if (armInteraction
+			&& event.getSource() == client.getLocalPlayer()
+			&& event.getTarget() instanceof NPC
+			&& ((NPC) event.getTarget()).getId() == NpcID.FISH_SHOAL)
+		{
+			tagFish(event.getTarget());
+			armInteraction = false;
+		}
+	}
+
+	@Subscribe
+	public void onGameTick(GameTick tick)
+	{
+		if (!inDriftNetArea)
+		{
+			return;
+		}
+
+		final int currentTickCount = client.getTickCount();
+		taggedFish.entrySet().removeIf(entry -> entry.getValue() + config.timeoutDelay() < currentTickCount);
+
+		armInteraction = false;
+	}
+
+	@Subscribe
+	public void onChatMessage(ChatMessage event)
+	{
+		if (!inDriftNetArea)
+		{
+			return;
+		}
+
+		if (event.getType() == ChatMessageType.SPAM && event.getMessage().equals(CHAT_PRODDING_FISH))
+		{
+			Actor target = client.getLocalPlayer().getInteracting();
+
+			if (target != null && target instanceof NPC && ((NPC) target).getId() == NpcID.FISH_SHOAL)
```

Comment:
`target instanceof NPC` is, by itself, a null check as well, hence explicitly null-checking isn't needed.
```suggestion
			if (target instanceof NPC && ((NPC) target).getId() == NpcID.FISH_SHOAL)
```